test/IdTripleTest.cpp

```diff
@@ -0,0 +1,59 @@
+// Copyright 2024, University of Freiburg
+//  Chair of Algorithms and Data Structures.
+//  Authors:
+//    2024 Julian Mundhahs <mundhahj@tf.uni-freiburg.de>
+
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+
+#include "./util/GTestHelpers.h"
+#include "./util/IdTestHelpers.h"
+#include "global/IdTriple.h"
+
+using namespace ad_utility::testing;
+
+TEST(IdTripleTest, constructors) {
+  const std::array<Id, 3> undefIds{Id::makeUndefined(), Id::makeUndefined(),
+                                   Id::makeUndefined()};
+  const std::array<Id, 3> ids{Id::makeFromInt(42), VocabId(10),
+                              Id::makeFromBool(false)};
+
+  {
+    auto idTriple = IdTriple();
+    EXPECT_THAT(idTriple, AD_FIELD(IdTriple<>, ids_,
+                                   testing::ElementsAreArray(undefIds)));
```

Comment:
How does this work? I don't think that the default constructor of ID creates deterministic values (for performance reasons).

Reply:
This should be caused by the difference between [value-](https://en.cppreference.com/w/cpp/language/value_initialization) und [default-initialization](https://en.cppreference.com/w/cpp/language/default_initialization) (e.g. `IdTriple idt();` or `IdTriple idt{};` vs. `IdTriple idt;`). If a non user provided default constructor is present value-initialization first does a zero-initialization and then a default-initialization.

Reply:
But as a default constructor without arguments is not required, I have removed it.